### PR TITLE
Add IStringLocalizer.GetAllStrings method

### PR DESCRIPTION
**Pull request by DamianEdwards**
@davidfowl @muratg 

#18 #23

### CODE REVIEW

File: src/Microsoft.Framework.Localization/ResourceManagerStringLocalizer.cs [hunk 1]
```diff
@@ -94,6 +94,28 @@ public IStringLocalizer WithCulture(CultureInfo culture)
...
+        /// <param name="culture">The <see cref="CultureInfo"/> to get strings for.</param>
+        /// <returns>The strings.</returns>
+        protected IEnumerable<LocalizedString> GetAllStrings([NotNull] CultureInfo culture)
+        {
+            var resourceNames = GetResourceNamesFromCultureHierarchy(culture);
+
+            foreach (var name in resourceNames)
+            {
+                var value = GetStringSafely(name, culture);
+                yield return new LocalizedString(name, value ?? name, resourceNotFound: value == null);
```
Comment (hishamco) [inline src/Microsoft.Framework.Localization/ResourceManagerStringLocalizer.cs]: According to the function definition https://github.com/aspnet/Localization/blob/dev/src/Microsoft.Framework.Localization.Abstractions/LocalizedString.cs#L30  I don't think that we need the named parameter ```resourceNotFound``` while you didn't change the order of the parameters

**Comment by muratg** [comment]
I think we should do https://github.com/aspnet/Localization/issues/23 in this PR.

File: src/Microsoft.Framework.Localization/ResourceManagerStringLocalizer.cs (hunk 1 above)
Comment (DamianEdwards) [inline src/Microsoft.Framework.Localization/ResourceManagerStringLocalizer.cs]: We always tend to name args when we pass in `bool` or `null` values.

**Comment by DamianEdwards** [comment]
@muratg I added changes for #23

**Comment by DamianEdwards** [comment]
@hishamco that's what the extension method achieves https://github.com/aspnet/Localization/pull/22/files#diff-fd6180fc301d656a5a25bc6d20b88a50R38

**Comment by hishamco** [comment]
:+1:

File: src/Microsoft.Framework.Localization.Abstractions/StringLocalizerExtensions.cs [hunk 2]
```diff
@@ -10,22 +11,31 @@ public static class StringLocalizerExtensions
...
+        public static LocalizedString GetString(
+            [NotNull] this IStringLocalizer stringLocalizer,
+            [NotNull] string name,
+            params object[] arguments) => stringLocalizer[name, arguments];
+
+        /// <summary>
+        /// Gets all string resources including those for ancestor cultures.
+        /// </summary>
+        /// <param name="stringLocalizer">The <see cref="IStringLocalizer"/>.</param>
+        /// <returns></returns>
```
Comment (kirthik) [inline src/Microsoft.Framework.Localization.Abstractions/StringLocalizerExtensions.cs]: `<returns>` should have some text. May be "The strings" like above?

**Comment by kirthik** [comment]
:shipit:

**Comment by muratg** [comment]
:shipit:

**Pull request merged**